src/yb/gutil/strings/numbers.cc

```diff
@@ -526,7 +526,7 @@ string Uint128ToHexString(uint128 ui128) {
 
 string Uint16ToHexString(uint16_t ui16) {
   char buf[5];
-  sprintf(buf, "%04X", ui16);
+  snprintf(buf, "%04X", ui16);
```

Comment:
You should pass the size limit as an argument.  One reason for using `snprintf` is to protect against buffer overflows.

Reply:
sorry I didn't recheck before push, still in local